ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -271,7 +272,7 @@ protected Boolean findNextCompactionAndExecute(boolean collectGenericStats, bool
       if (ci == null) {
         return false;
       }
-      if ((runtimeVersion != null || ci.initiatorVersion != null) && !runtimeVersion.equals(ci.initiatorVersion)) {
+      if ((runtimeVersion != null && ci.initiatorVersion != null) && !runtimeVersion.equals(ci.initiatorVersion)) {
```

Comment:
What happens if `runtimeVersion` is not null but `ci.initiatorVersion` is null? Shouldn't we log the mismatch in this case?

Reply:
with the old criteria is runtimeVersion is null and initiatorVersion is not null, it will lead to an NPE. This change is unrelated to the PR, just wanted to fix the issue above. However you are right this does not cover all cases.